src/main.rs

```diff
@@ -0,0 +1,37 @@
+//I'd like the most pedantic warning level
+#![warn(clippy::pedantic, clippy::needless_borrow)]
+// But I don't care about these ones for now (most applicable since the code isn't fleshed out)
+#![allow(clippy::module_name_repetitions)]
+#![allow(clippy::needless_pass_by_value)]
```

Comment:
Update to this:
```
// I'd like the most pedantic warning level
#![warn(
    clippy::cargo,
    clippy::needless_borrow,
    clippy::pedantic,
    clippy::redundant_clone
)]
// But I don't care about these ones
#![allow(
    clippy::cast_precision_loss,     // There is no way to avoid this precision loss
    clippy::module_name_repetitions, // Sometimes clear naming calls for repetition
    clippy::multiple_crate_versions,  // There is no way to easily fix this without modifying our dependencies
    clippy::needless_pass_by_value // FIXME: Remove once the code is in a better state 
)]
```
Then rerun `cargo clippy`